controllers/authenticationController.js

```diff
@@ -0,0 +1,259 @@
+const jwt = require('jsonwebtoken');
+const crypto = require('crypto');
+const bcrypt = require('bcryptjs');
+
+const userModel = require('../models/userModel');
+const doctorModel = require('../models/doctorModel');
+const adminModel = require('../models/adminModel');
+const AppError = require('../utils/appError');
+const sendEmail = require('../utils/sendEmail');
+
+const correctPassword = async function (candidatePassword, userPassword) {
+  return await bcrypt.compare(candidatePassword, userPassword);
+};
+const getSignToken = (id) => {
+  const token = jwt.sign({ id: id }, process.env.JWT_SECRET, {
+    expiresIn: process.env.JWT_EXPIRES_IN,
+  });
+  return token;
+};
+const isCorrectPassword = async function (candidatePassword, passwordInDb) {
+  return await bcrypt.compare(candidatePassword, passwordInDb);
+};
+const IsChangedPasswordAfterGetToken = async function (
+  jwtTimeIat,
+  currentUser
+) {
+  if (currentUser.passwordChangedAt) {
+    const passwordChangedAtInSec = parseInt(
+      currentUser.passwordChangedAt.getTime() / 1000,
+      10
+    );
+
+    return jwtTimeIat < passwordChangedAtInSec;
+  }
+
+  // False means NOT changed
+  return false;
+};
+const createPasswordReset = function (currentUser) {
+  const resetToken = crypto.randomBytes(32).toString('hex');
+
+  currentUser.passwordResetToken = crypto
+    .createHash('sha256')
+    .update(resetToken)
+    .digest('hex');
+
+  //resetToken Expires after 10miute
+  currentUser.passwordResetExpires = Date.now() + 10 * 60 * 1000;
+
+  return resetToken;
+};
+exports.signUp = async (req, res, next) => {
+  try {
+    let newUser;
+    if (req.baseUrl === '/psy/doctor') {
```

Comment:
Here you better send the user type in req.body and check it instead of checking the baseUrl